lib/temporal/errors.rb

```diff
@@ -25,6 +25,24 @@ class WorkflowNotRegistered < ClientError; end
   class ApiError < Error; end
 
   class NotFoundFailure < ApiError; end
+
+  # Superclass for errors raised when retrieving a workflow result on the
```

Comment:
nit: These are technically broader than what's described here. A terminated workflow, for example, doesn't involve any failure on the worker.